File: appium/sample-scripts/ruby/testdroid_ios.rb

# install the required gems with bundler by doing:
#   "bundle install"
require 'json'
require 'rspec'
require 'selenium-webdriver'
require 'curb'
require 'selenium/webdriver/remote/http/curb'
include Selenium


##
## IMPORTANT: Set the following parameters.
##
screen_shot_dir= "screenshot-folder"
testdroid_username = ENV["TESTDROID_USERNAME"]
testdroid_password = ENV["TESTDROID_PASSWORD"]
testdroid_device = "iPad 3 A1416 7.0.4" # Example device. Change if you desire.
testdroid_app_file = "BitbarIOSSample.ipa" 


def log(msg)
    puts "#{Time.now}: #{msg}"
end
@testdroid_app=nil
desired_capabilities_cloud={
        'app'=> 'com.bitbar.testdroid.BitbarIOSSample',
        'device'=> 'iphone',
        'testdroid_app'=> nil,
        'testdroid_username'=> testdroid_username,
        'testdroid_password'=> testdroid_password,
        'testdroid_project'=> 'Appium iOS Project1',
        'testdroid_description'=> 'Appium project description',
        'testdroid_testrun'=> 'Test Run 1',
        'testdroid_device'=> testdroid_device,
    }


server_url = 'http://appium.testdroid.com/wd/hub'

def upload_application(file_path, username, password)
  
  c = Curl::Easy.new("http://appium.testdroid.com/upload")
  c.http_auth_types = :basic
  c.username = username
  c.password = password
  c.multipart_form_post = true
  c.verbose = true
  c.http_post(Curl::PostField.file('BitbarIOSSample.ipa', file_path))
  resp = JSON.parse(c.body_str)

  @testdroid_app = resp["value"]["uploads"]["BitbarIOSSample.ipa"]
end

describe "BitbarIOSSample testing" do
  before :all do
 
    log ("Upload application #{testdroid_app_file}")
    upload_application(testdroid_app_file, testdroid_username , testdroid_password)
    log ("Uploaded file uuid #{@testdroid_app}")

    desired_capabilities_cloud['testdroid_app']=@testdroid_app
    http_client = WebDriver::Remote::Http::Curb.new
    http_client.timeout = nil #not timeout for Webdriver calls
    log ("Start Wbdriver with [#{desired_capabilities_cloud}]")
    @driver = Selenium::WebDriver.for(:remote, :desired_capabilities => desired_capabilities_cloud, :url => server_url, :http_client => http_client)

    log ("WebDriver response received")
   end

  after :all do
    log ("Stop WebDriver")

    @driver.quit
  end

  it "should show failure page"  do
    log ("view1: Finding buttons")
    buttons = @driver.find_elements(:tag_name, :button)
    log ("view1: Clicking button [0] - RadioButton 1")
    buttons[0].click()

    log ("view1: Typing in textfield[0]: Testdroid user")
    @driver.find_elements(:tag_name,  :textField)[0].send_keys("Testdroid user")

    y = 0.95
    x = 0.5
    log ("view1: Tapping at position (384, 0.95) - Estimated position of SpaceBar")
    @driver.execute_script("mobile: tap",{"touchCount" => "1","x" => x,"y" => y})

    log ("view1: Taking screenshot screenshot1.png")
    @driver.save_screenshot(screen_shot_dir + "/screenshot1.png")
    
    log ("view1: Hiding Keyboard")
    @driver.execute_script('UIATarget.localTarget().frontMostApp().keyboard().buttons()["Hide keyboard"].tap();')

    log ("view1: Taking screenshot screenshot2.png")
    @driver.save_screenshot(screen_shot_dir + "/screenshot2.png")

    log ("view1: Clicking button[6] - OK  Button")
    buttons[6].click()

    log ("view2: Taking screenshot screenshot3.png")
    @driver.save_screenshot(screen_shot_dir + "/screenshot3.png")
  end

  it "should click back button" do
     # view2
    log ("view2: Finding buttons")
    buttons = @driver.find_elements(:tag_name, :button)
    log ("view2: Clicking button[0] - Back/OK button")
    buttons[0].click()
  end

  it "should click 2nd radio button" do
    log ("view1: Finding buttons")
    buttons = @driver.find_elements(:tag_name, :button)
    log ("view1: Clicking button[2] - RadioButton 2")
    buttons[2].click()

    log ("view1: Clicking button[6] - OK Button")
    buttons[6].click()
    log ("view1: Taking screenshot screenshot4.png")
    @driver.save_screenshot(screen_shot_dir + "/screenshot4.png")

    log ("view1: Sleeping 3 before quitting webdriver")
    sleep(3)

  end

  it "should get window size" do
    size = @driver.manage.window.size
    size.width.should eq(768)
    size.height.should eq(1024)
  end

end

File: appium/sample-scripts/python/testdroid_ios.py
##
## For help on setting up your machine and configuring this TestScript go to
## http://docs.bitbar.com/testing/appium/
##

import os
import time
import unittest
from time import sleep
from appium import webdriver
from device_finder import DeviceFinder


def log(msg):
    print (time.strftime("%H:%M:%S") + ": " + msg)


class TestdroidIOS(unittest.TestCase):
    """
    Take screenshot and store files to defined location, with numbering prefix

    :Args:
    - name - files are stored as #_name
    """

    def screenshot(self, name):
        screenshot_name = str(self.screenshot_count) + "_" + name + ".png"
        log("Taking screenshot: " + screenshot_name)
        self.driver.save_screenshot(self.screenshot_dir + "/" + screenshot_name)
        self.screenshot_count += 1

    def setUp(self):

        ##
        ## IMPORTANT: Set the following parameters.
        ##
        testdroid_url = os.environ.get('TESTDROID_URL') or "https://cloud.bitbar.com"
        appium_url = os.environ.get('TESTDROID_APPIUM_URL') or 'https://appium.bitbar.com/wd/hub'
        testdroid_apiKey = os.environ.get('TESTDROID_APIKEY') or ""
        testdroid_project_name = os.environ.get('TESTDROID_PROJECT') or "iOS sample project"
        testdroid_testrun_name = os.environ.get('TESTDROID_TESTRUN') or "My testrun"
        testdroid_app = os.environ.get('TESTDROID_APP') or ""
        testdroid_bundle_id = os.environ.get('TESTDROID_BUNDLE_ID') or "com.bitbar.testdroid.BitbarIOSSample"
        new_command_timeout = os.environ.get('TESTDROID_CMD_TIMEOUT') or '60'
        testdroid_test_timeout = os.environ.get('TESTDROID_TEST_TIMEOUT') or '600'
        testdroid_find_device = os.environ.get('TESTDROID_FINDDEVICE') or "true"
        automation_name = os.environ.get('TESTDROID_AUTOMATION_NAME') or "XCUITest"

        self.screenshot_dir = os.environ.get('TESTDROID_SCREENSHOTS') or os.getcwd() + "/screenshots"
        log("Will save screenshots at: " + self.screenshot_dir)
        self.screenshot_count = 1

        # Options to select device
        # 1) Set environment variable TESTDROID_DEVICE
        # 2) Set device name to this python script
        # 3) Do not set #1 and #2 and let DeviceFinder to find free device for you
        testdroid_device = os.environ.get('TESTDROID_DEVICE') or ""

        deviceFinder = DeviceFinder(url=testdroid_url)
        if testdroid_device == "":
            # Loop will not exit until free device is found
            while testdroid_device == "":
                testdroid_device = deviceFinder.available_ios_device()

        print "Starting Appium test using device '%s'" % testdroid_device

        desired_capabilities_cloud = {}
        desired_capabilities_cloud['testdroid_apiKey'] = testdroid_apiKey
        desired_capabilities_cloud['testdroid_target'] = 'ios'
        desired_capabilities_cloud['testdroid_project'] = testdroid_project_name
        desired_capabilities_cloud['testdroid_testrun'] = testdroid_testrun_name
        desired_capabilities_cloud['testdroid_device'] = testdroid_device
        desired_capabilities_cloud['testdroid_app'] = testdroid_app
        desired_capabilities_cloud['platformName'] = 'iOS'
        desired_capabilities_cloud['deviceName'] = 'iPhone device'
        desired_capabilities_cloud['newCommandTimeout'] = new_command_timeout
        desired_capabilities_cloud['testdroid_testTimeout'] = testdroid_test_timeout
        desired_capabilities_cloud['testdroid_findDevice'] = testdroid_find_device
        desired_capabilities_cloud['automationName'] = automation_name
        desired_capabilities_cloud['app'] = testdroid_bundle_id

        # set up webdriver
        log("WebDriver request initiated. Waiting for response, this typically takes 2-3 mins")

        self.driver = webdriver.Remote(command_executor=appium_url, desired_capabilities=desired_capabilities_cloud)
        log("WebDriver response received")

    def tearDown(self):
        log("Quitting")
        self.driver.quit()

    def testSample(self):
        # view1
        log("view1: Finding buttons")
        buttons = self.driver.find_elements_by_class_name('UIAButton')
        log("view1: Clicking button [0] - RadioButton 1")
        buttons[0].click()

        log("view1: Typing in textfield[0]: Testdroid user")
        elem = self.driver.find_element_by_class_name('UIATextField')
        elem.clear()
        elem.send_keys('Testdroid user')

        log("view1: Taking screenshot screenshot1.png")
        self.screenshot("screenshot1")

        log("view1: Hiding Keyboard")
        self.driver.find_element_by_xpath("//*[contains(@name, 'Return')]").click()

        log("view1: Taking screenshot screenshot2.png")
        self.screenshot("screenshot2")

        log("view1: Clicking button[6] - OK  Button")
        buttons[6].click()

        log("view2: Taking screenshot screenshot3.png")
        self.screenshot("screenshot3")

        # view2
        log("view2: Finding buttons")
        buttons = self.driver.find_elements_by_class_name('UIAButton')
        log("view2: Clicking button[0] - Back/OK button")
        buttons[0].click()

        # view1
        log("view1: Finding buttons")
        buttons = self.driver.find_elements_by_class_name('UIAButton')
        log("view1: Clicking button[2] - RadioButton 2")
        buttons[2].click()

        log("view1: Clicking button[6] - OK Button")
        buttons[6].click()

        log("view1: Taking screenshot screenshot4.png")
        self.screenshot("screenshot4")

        log("view1: Sleeping 3 before quitting webdriver.")
        sleep(3)


def initialize():
    return TestdroidIOS


if __name__ == "__main__":
    suite = unittest.TestLoader().loadTestsFromTestCase(TestdroidIOS)
    unittest.TextTestRunner(verbosity=2).run(suite)
